pilot/pkg/xds/delta.go

```diff
@@ -284,11 +288,7 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
 		return nil
 	}
 
-	subs := sets.New(req.ResourceNamesSubscribe...).Delete("*")
-	// InitialResourceVersions are essential subscriptions on the first request, since we don't care about the version
-	for k := range req.InitialResourceVersions {
-		subs.Insert(k)
-	}
+	subs, _ := deltaWatchedResources(nil, req)
```

Comment:
Does this function remove * from the set?

Reply:
right. It removes *

>Not sure what sub+unsub in one request means, though. So probably fine

I cannot find the history added it there, I guess that is only for resilience.